flexbox/flex-basics.md

```diff
@@ -0,0 +1,245 @@
+
+#### Column
+
+Will align children elements if the parent container in a column.
+
+```css
+.parent {
+  display: flex;
+  flex-direction: column;
+}
+```
+![alt text](https://i.imgur.com/8mUkA4R.png "flex-direction: column;")
+
+#### Column-Reverse
+
+Functions the same as column with the exception of ordering the child elements in reverse.
+
+```css
+.parent {
+  display: flex;
+  flex-direction: column-reverse;
+}
+```
+![alt text](https://i.imgur.com/MwDhG1m.png "flex-direction: column-reverse;")
+
+## Alignment
+
+### Justify Content
+
+This property allows for the alignment of children along the primary axis of the flex container.  So if we have a parent container with row we will be justifying content along the X-Axis while if it were a column we'd be justifying it along the Y-Axis.
+
+For the purpose of the following examples, the flex-direction will be a row but bear in mind the structuring applies the same for columns.
+
+#### Flex-Start
+
+Will justify the content among the beginning of the flex axis, so being a row we will "float the contents" to the start or left of the axis.
+
+```css
+.parent {
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+}
+```
+![alt text](https://i.imgur.com/SbcPZye.png "justify-content: flex-start;")
+
+#### Flex-End
+
+I'm sure you're noticing some patterns by now.  Much like flex-start, flex-end will float the children elements to the end of the axis.  Since the flex-direction is the row that puts our little red buddies all the way to the right.
+
+```css
+.parent {
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
+}
+```
+![alt text](https://i.imgur.com/lJX7RpO.png "justify-content: flex-end;")
+
+#### Center
+
+Justify-Content: center; much like the rest of flexbox attribute values works exactly as it sounds.  It will center the contents of the container within the center of its flex-direction's axis.
+
+```css
+.parent {
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+}
+```
+![alt text](https://i.imgur.com/4039Flq.png "justify-content: center;")
+
+#### Space Between
+
+Space between will adjust the content evenly between each child elements. Bear in mind if the parent container has padding it won't be set to the edge like in the example below.
+
+```css
+.parent {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+}
+```
+![alt text](https://i.imgur.com/c6XdOql.png "justify-content: space-between;")
+
+#### Space Around
+
+Space around will adjust the content of the children elements automatically based on the space around them (not just between). Again the padding of the container and margin of the children will affect the final result of space-around.
+
+```css
+.parent {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-around;
+}
+```
+![alt text](https://i.imgur.com/Gzc2uEn.png "justify-content: space-around;")
+
+## Sources and External Resources
```

Comment:
Another great resource, if you want to add it: WesBos has a free, 20-minute tutorial called [What The Flexbox?!](https://flexbox.io/). His courses are kind of code-along style, with example files you can fork and clone on GitHub.
[This](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) is another great resource too :)